penguin/llm/adapters/anthropic.py

```
@@ -818,17 +918,130 @@ def process_response(self, response: Any) -> Tuple[str, List[Any]]:
+        if message.get("role") != "assistant":
+            return set()
+        content = message.get("content")
+        if not isinstance(content, list):
+            return set()
+        return {
+            str(part.get("id"))
+            for part in content
+            if isinstance(part, dict) and part.get("type") == "tool_use"
+        }
+
     def format_messages(self, messages: List[Dict[str, Any]]) -> List[Dict[str, Any]]:
```

Ruff (0.15.12)

[warning] Too many branches (50 > 12)

(PLR0912)